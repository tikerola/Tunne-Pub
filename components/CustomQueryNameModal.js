import React from "react";
import { StyleSheet, View, Text, TextInput, Modal } from "react-native";
import { colors } from "../constants/colors";
import CustomButton from "./CustomButton";

export default function CustomQueryNameModal({
  navigation,
  modalVisible,
  setModalVisible,
  summaryItemName,
  error,
  setSummaryItemName,
  handleModalPress,
  pad = false,
}) {
  return (
    <View testID="modal-container" style={styles.modalContainer}>
      <Modal
        transparent={true}
        animationType="fade"
        hardwareAccelerated
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(false);
          navigation.navigate("Ohjeet");
        }}
      >
        <View style={styles.modalContentContainer}>
          <View style={{ alignSelf: "center", width: pad ? "65%" : "100%" }}>
            <View>
              <Text testID="modal-title" style={styles.descText}>
                Aloita antamalla istunnolle nimi
              </Text>
              <TextInput
                testID="input"
                style={styles.textInput}
                onChangeText={(text) => setSummaryItemName(text)}
                value={summaryItemName}
              />
              <Text style={styles.error}>{error}</Text>
            </View>
            <View style={styles.buttonContainer}>
              <CustomButton title="Tallenna" onPress={handleModalPress} />
              <CustomButton
                title="Peruuta"
                onPress={() => handleModalPress("peruuta")}
                color="red"
              />
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
}

const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "flex-start",
    backgroundColor: colors.bodyBg,
  },

  modalContentContainer: {
    display: "flex",
    flexDirection: "column",
    flex: 1,
    padding: 25,
    justifyContent: "center",
    shadowColor: "#fff",
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

    elevation: 5,
  },

  textInput: {
    borderWidth: 1,
    backgroundColor: "#555",
    borderRadius: 5,
    padding: 5,
    borderColor: "#bbb",
    color: "white",
  },
  descText: {
    fontSize: 18,
    fontFamily: "Roboto-Light",
    color: "white",
    marginBottom: 5,
  },

  error: {
    textAlign: "center",
    fontSize: 16,
    fontFamily: "Roboto-Bold",
    color: "red",
  },
  pad: {
    backgroundColor: colors.stepperContentBg,
    padding: 50,
    borderRadius: 10,
  },
  buttonContainer: {
    marginTop: 5,
    flexDirection: "row",
    justifyContent: "space-around",
  },
});
