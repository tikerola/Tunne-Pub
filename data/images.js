import image1 from "../assets/images/1.jpg";
import image2 from "../assets/images/2.jpg";
import image3 from "../assets/images/3.jpg";
import image4 from "../assets/images/4.jpg";
import image5 from "../assets/images/5.jpg";
import image6 from "../assets/images/6.jpg";
import image7 from "../assets/images/7.jpg";
import image8 from "../assets/images/8.jpg";
import image9 from "../assets/images/9.jpg";
import image10 from "../assets/images/10.jpg";
import image11 from "../assets/images/11.jpg";
import image12 from "../assets/images/12.jpg";
import image13 from "../assets/images/13.jpg";
import image14 from "../assets/images/14.jpg";
import image15 from "../assets/images/15.jpg";
import image16 from "../assets/images/16.jpg";
import image17 from "../assets/images/17.jpg";
import image18 from "../assets/images/18.jpg";
import image19 from "../assets/images/19.jpg";
import image20 from "../assets/images/20.jpg";
import image21 from "../assets/images/21.jpg";
import image22 from "../assets/images/22.jpg";
import image23 from "../assets/images/23.jpg";
import image24 from "../assets/images/24.jpg";
import image25 from "../assets/images/25.jpg";
import image26 from "../assets/images/26.jpg";
import image27 from "../assets/images/27.jpg";
import image28 from "../assets/images/28.jpg";
import image29 from "../assets/images/29.jpg";
import image30 from "../assets/images/30.jpg";
import image31 from "../assets/images/31.jpg";
import image32 from "../assets/images/32.jpg";
import image33 from "../assets/images/33.jpg";
import image34 from "../assets/images/34.jpg";
import image35 from "../assets/images/35.jpg";
import image36 from "../assets/images/36.jpg";
import image37 from "../assets/images/37.jpg";
import image38 from "../assets/images/38.jpg";
import image39 from "../assets/images/39.jpg";
import image40 from "../assets/images/40.jpg";
import image41 from "../assets/images/41.jpg";
import image42 from "../assets/images/42.jpg";
import image43 from "../assets/images/43.jpg";
import image44 from "../assets/images/44.jpg";
import image45 from "../assets/images/45.jpg";
import image46 from "../assets/images/46.jpg";
import image47 from "../assets/images/47.jpg";
import image48 from "../assets/images/48.jpg";
import image49 from "../assets/images/49.jpg";

export {
  image1,
  image2,
  image3,
  image4,
  image5,
  image6,
  image7,
  image8,
  image9,
  image10,
  image11,
  image12,
  image13,
  image14,
  image15,
  image16,
  image17,
  image18,
  image19,
  image20,
  image21,
  image22,
  image23,
  image24,
  image25,
  image26,
  image27,
  image28,
  image29,
  image30,
  image31,
  image32,
  image33,
  image34,
  image35,
  image36,
  image37,
  image38,
  image39,
  image40,
  image41,
  image42,
  image43,
  image44,
  image45,
  image46,
  image47,
  image48,
  image49,
};
