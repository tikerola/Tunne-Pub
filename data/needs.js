const NEEDS_DATA = [
  { id: '1', word: 'Arvostus' },
  { id: '2', word: 'empatia' },
  { id: '3', word: 'hellyys' },
  { id: '4', word: 'henkinen turva' },
  { id: '5', word: 'merkityksellisyys' },
  { id: '6', word: 'huolenpito' },
  { id: '7', word: 'hoiva' },
  { id: '8', word: 'huomioiduksi tuleminen' },
  { id: '9', word: 'hyväksyntä' },
  { id: '10', word: 'Johdonmukaisuus' },
  { id: '11', word: 'kiintymys' },
  { id: '12', word: 'kumppanuus' },
  { id: '13', word: 'kunnioitus' },
  { id: '14', word: 'luottamus' },
  { id: '15', word: 'läheisyys' },
  { id: '16', word: 'myötätunto' },
  { id: '17', word: 'rakkaus' },
  { id: '18', word: 'nähdyksi tuleminen' },
  { id: '19', word: 'kuulluksi tuleminen' },
  { id: '20', word: 'Vastavuoroisuus' },
  { id: '21', word: 'vuorovaikutus' },
  { id: '22', word: 'ymmärretyksi tuleminen' },
  { id: '23', word: 'Hauskanpito' },
  { id: '24', word: 'huumori' },
  { id: '25', word: 'seikkailu' },
  { id: '26', word: 'Aitous' },
  { id: '27', word: 'rehellisyys' },
  { id: '28', word: 'läsnäolo' },
  { id: '29', word: 'Tieto' },
  { id: '30', word: 'oma tila' },
  { id: '31', word: 'vaihtelu' },
  { id: '32', word: 'vapaus' },
  { id: '33', word: 'vastuu' },
  { id: '34', word: 'yksityisyys' },
  { id: '35', word: 'ennakoitavuus' },
  { id: '36', word: 'kauneus' },
  { id: '37', word: 'Lohtu' },
  { id: '38', word: 'reiluus' },
  { id: '39', word: 'tasavertaisuus' },
  { id: '40', word: 'suojeleminen' },
  { id: '41', word: 'tasapaino' },
  { id: '42', word: 'yhteisöllisyys' },
  { id: '43', word: 'Kosketus' },
  { id: '44', word: 'lepo' },
  { id: '45', word: 'uni' },
  { id: '46', word: 'liikkuminen' },
  { id: '47', word: 'nautinto' },
  { id: '48', word: 'ravinto' },
  { id: '49', word: 'rentoutuminen' },
  { id: '50', word: 'seksuaalinen ilmaisu' },
  { id: '51', word: 'Aikaansaaminen' },
  { id: '52', word: 'haasteellisuus' },
  { id: '53', word: 'kiitollisuus' },
  { id: '54', word: 'hengellisyys' },
  { id: '55', word: 'itseilmaisu' }
]

export default NEEDS_DATA





