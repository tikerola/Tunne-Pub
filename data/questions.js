







export const QUESTIONS = [
  'Sinulla on ollut jokin tunteitanostattava tilanne? Kerro mitä tapahtui.',
  'Sanat, jotka parhaiten kuvaavat tunteitasi',
  'Kuvat, jotka parhaiten kuvaavat tunteitasi.',
  'Missä ja miten tunne tuntuu kehossa (esim. paino rinnan päällä, pala kurkussa, kädet menee nyrkkiin)',
  'Onko tunne alunperin sinun vai nappasitko jonkun toisen tunteen?',
  'Mikä tarve tunteen takana? Mitä olisit siinä hetkessä tarvinnut? (kuulluksi tulemista, lohdutusta, jne)',
  'Mitä tunnetta olisit kaivannut?',
  'Mikä tarve sinulla on juuri nyt?',
  'Mitä lausetta pyörität päässäsi juuri nyt? Mikä uskomus sinulla on? Mitä se sinulle merkitsee?',
  'Mitä sinun nyt pitäisi tehdä jotta voisit hyvin?'
]