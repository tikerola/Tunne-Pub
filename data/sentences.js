export const SENTENCES_DATA = [
  { id: '1', word: 'mitä muut ajattelevat?' },
  { id: '2', word: 'Olen niin huono!' },
  { id: '3', word: 'Olen väärässä.' },
  { id: '4', word: 'Joku muu tietää paremmin.' },
  { id: '5', word: 'Olen aina ennenkin antanut periksi.' },
  { id: '6', word: 'Ei tästä mitään tule, minulla ei ole rohkeutta.' },
  { id: '7', word: 'En minä osaa...' },
  { id: '8', word: 'Toinen suuttuu kuitenkin.' },
  { id: '9', word: 'Pelkään tuottaa pettymystä toiselle.' },
  { id: '10', word: 'Parempi vaan nyt olla hiljaa. Olen eri mieltä kuin muut ja pelkään tulla mielipiteeni kanssa näkyväksi.' },
  { id: '11', word: 'Tunteeni, tarpeeni, ajatukseni on huono/hölmö tai tyhmä.' }
]