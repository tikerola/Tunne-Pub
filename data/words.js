const WORD_DATA = [
  { id: '1', word: 'avuton' },
  { id: '2', word: 'ahdistunut' },
  { id: '3', word: 'alistunut' },
  { id: '4', word: 'arka' },
  { id: '5', word: 'epäilevä' },
  { id: '6', word: 'epäonnistunut' },
  { id: '7', word: 'epätoivoinen' },
  { id: '8', word: 'haluton' },
  { id: '9', word: 'häpeissään' },
  { id: '10', word: 'hämmentynyt' },
  { id: '11', word: 'huojentunut' },
  { id: '12', word: 'huolissaan' },
  { id: '13', word: 'iloinen' },
  { id: '14', word: 'innokas' },
  { id: '15', word: 'itsevarma' },
  { id: '16', word: 'jännittynyt' },
  { id: '17', word: 'kauhuissaan' },
  { id: '18', word: 'kiitollinen' },
  { id: '19', word: 'kaihoisa' },
  { id: '20', word: 'kateellinen' },
  { id: '21', word: 'katkera' },
  { id: '22', word: 'kyyninen' },
  { id: '23', word: 'kyllästynyt' },
  { id: '24', word: 'lamaantunut' },
  { id: '25', word: 'levollinen' },
  { id: '26', word: 'masentunut' },
  { id: '27', word: 'malttamaton' },
  { id: '28', word: 'onnellinen' },
  { id: '29', word: 'pelokas' },
  { id: '30', word: 'pettynyt' },
  { id: '31', word: 'päättäväinen' },
  { id: '32', word: 'puolusteleva' },
  { id: '33', word: 'rohkea' },
  { id: '34', word: 'rakastunut' },
  { id: '35', word: 'surullinen' },
  { id: '36', word: 'syyllinen' },
  { id: '37', word: 'tarmokas' },
  { id: '38', word: 'toiveikas' },
  { id: '39', word: 'uupunut' },
  { id: '40', word: 'utelias' },
  { id: '41', word: 'vaativa' },
  { id: '42', word: 'voittajafiilis' },
  { id: '43', word: 'väkinäinen' },
  { id: '44', word: 'vastenmielinen' },
  { id: '45', word: 'vapautunut' },
  { id: '46', word: 'vihainen' },
  { id: '47', word: 'ylpeä' },
  { id: '48', word: 'ylimielinen' },
  { id: '49', word: 'yllättynyt' },
  { id: '50', word: 'ärtynyt' }
]

export default WORD_DATA