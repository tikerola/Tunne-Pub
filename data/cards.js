import Card from "../models/card";
import * as images from "./images";

const CARD_DATA = [
  new Card("1", "happy", images.image1),
  new Card("2", "sad", images.image2),
  new Card("3", "worried", images.image3),
  new Card("4", "scared", images.image4),
  new Card("5", "happy", images.image5),
  new Card("6", "sad", images.image6),
  new Card("7", "worried", images.image7),
  new Card("8", "scared", images.image8),

  new Card("9", "happy", images.image9),
  new Card("10", "sad", images.image10),
  new Card("11", "worried", images.image11),
  new Card("12", "scared", images.image12),
  new Card("13", "happy", images.image13),
  new Card("14", "sad", images.image14),
  new Card("15", "worried", images.image15),
  new Card("16", "scared", images.image16),

  new Card("17", "happy", images.image17),
  new Card("18", "sad", images.image18),
  new Card("19", "worried", images.image19),
  new Card("20", "scared", images.image20),
  new Card("21", "happy", images.image21),
  new Card("22", "sad", images.image22),
  new Card("23", "worried", images.image23),
  new Card("24", "scared", images.image24),

  new Card("25", "happy", images.image25),
  new Card("26", "sad", images.image26),
  new Card("27", "worried", images.image27),
  new Card("28", "scared", images.image28),
  new Card("29", "happy", images.image29),
  new Card("30", "sad", images.image30),
  new Card("31", "worried", images.image31),
  new Card("32", "scared", images.image32),

  new Card("33", "happy", images.image33),
  new Card("34", "sad", images.image34),
  new Card("35", "worried", images.image35),
  new Card("36", "scared", images.image36),
  new Card("37", "happy", images.image37),
  new Card("38", "sad", images.image38),
  new Card("39", "worried", images.image39),
  new Card("40", "scared", images.image40),

  new Card("41", "happy", images.image41),
  new Card("42", "sad", images.image42),
  new Card("43", "worried", images.image43),
  new Card("44", "scared", images.image44),
  new Card("45", "happy", images.image45),
  new Card("46", "sad", images.image46),
  new Card("47", "worried", images.image47),
  new Card("48", "scared", images.image48),
  new Card("49", "scared", images.image49),
];

export default CARD_DATA;
