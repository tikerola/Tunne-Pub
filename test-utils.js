import React from "react";
import { NavigationContext } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CARD_DATA from "./data/cards";
import NEEDS_DATA from "./data/needs";
import { SENTENCES_DATA } from "./data/sentences";
import WORD_DATA from "./data/words";

const INITIAL_STATE = {
  cards: {
    // cards: CARD_DATA,
    // words: WORD_DATA,
    // needs: NEEDS_DATA,
    // sentences: SENTENCES_DATA,
    activeSummaryItemId: "123",
    summaryOfChosenItems: {
      123: {
        id: "123",
        name: "",
        cards: [],
        feelingsWords: [],
        feelingsYouMissWords: [],
        needs: ["hei"],
        needsBehindAFeeling: [],
        sentences: [],
        questions: {},
        itemSaved: true,
      },
    },
    screenNumber: 1,
    numOfItems: 0,
    initialLoadFinished: false,
  },
};

const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};

const reduxRender = (ui, options, storeData = store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={storeData}>
        <NavigationContext.Provider value={navContext}>
          {children}
        </NavigationContext.Provider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from "@testing-library/react-native";

// override render method
export { reduxRender as render, store };
