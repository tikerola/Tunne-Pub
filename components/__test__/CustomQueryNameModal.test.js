import { fireEvent } from "@testing-library/react-native";
import React from "react";
import configureStore from "redux-mock-store";
import { QUESTIONS } from "../../data/questions";
import { render } from "../../test-utils";
import CustomQueryNameModal from "../CustomQueryNameModal";

describe("<CustomQueryNameModal />", () => {
  const INITIAL_STATE = {
    cards: {
      cards: [],
      activeSummaryItemId: "123",
      summaryOfChosenItems: {
        123: {
          id: "123",
          name: "my name",
          cards: [],
          feelingsWords: ["pahoillaan", "yskä"],
          feelingsYouMissWords: [],
          needs: ["hei"],
          needsBehindAFeeling: [],
          sentences: [],
          questions: {
            [QUESTIONS[7]]: "niinpä niin",
            [QUESTIONS[8]]: "pahoillaan, yskä",
            [QUESTIONS[9]]: "diipadaapa",
          },
          itemSaved: true,
        },
      },
      screenNumber: 1,
      numOfItems: 0,
      initialLoadFinished: false,
    },
  };

  const navigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  modalVisible = true;
  setModalVisible = jest.fn();
  summaryItemName = "my name";
  error = "some error";
  setSummaryItemName = jest.fn();
  handleModalPress = jest.fn();

  const mockStore = configureStore([]);
  const store = mockStore(INITIAL_STATE);

  it("should match snapshot", () => {
    expect(
      render(
        <CustomQueryNameModal
          navigation={navigation}
          modalVisible={modalVisible}
          setModalVisible={setModalVisible}
          summaryItemName={summaryItemName}
          error={error}
          setSummaryItemName={setSummaryItemName}
          handleModalPress={handleModalPress}
        />,
        undefined,
        store
      )
    ).toMatchSnapshot();
  });

  it("should hide modal when pressed cancel", () => {
    const component = render(
      <CustomQueryNameModal
        navigation={navigation}
        modalVisible={modalVisible}
        setModalVisible={setModalVisible}
        summaryItemName={summaryItemName}
        error={error}
        setSummaryItemName={setSummaryItemName}
        handleModalPress={handleModalPress}
      />,
      undefined,
      store
    );

    fireEvent.press(component.queryByText("Peruuta"));
    expect(handleModalPress).toHaveBeenCalledWith("peruuta");
  });
});
