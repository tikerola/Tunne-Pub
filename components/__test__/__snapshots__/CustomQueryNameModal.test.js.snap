// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomQueryNameModal /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5f0c38",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
  testID="modal-container"
>
  <Modal
    animationType="fade"
    hardwareAccelerated={true}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "display": "flex",
          "elevation": 5,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 25,
          "shadowColor": "#fff",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.44,
          "shadowRadius": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Roboto-Light",
                "fontSize": 18,
                "marginBottom": 5,
              }
            }
            testID="modal-title"
          >
            Aloita antamalla istunnolle nimi
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#555",
                "borderColor": "#bbb",
                "borderRadius": 5,
                "borderWidth": 1,
                "color": "white",
                "padding": 5,
              }
            }
            testID="input"
            underlineColorAndroid="transparent"
            value="my name"
          />
          <Text
            style={
              Object {
                "color": "red",
                "fontFamily": "Roboto-Bold",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            some error
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": 5,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#3a89ed",
                "borderRadius": 5,
                "elevation": 5,
                "margin": 0,
                "opacity": 1,
                "padding": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.44,
                "shadowRadius": 5,
                "width": 100,
              }
            }
            testID="button"
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "Roboto-Light",
                  "fontSize": 16,
                }
              }
              testID="title"
            >
              Tallenna
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "red",
                "borderRadius": 5,
                "elevation": 5,
                "margin": 0,
                "opacity": 1,
                "padding": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.44,
                "shadowRadius": 5,
                "width": 100,
              }
            }
            testID="button"
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "Roboto-Light",
                  "fontSize": 16,
                }
              }
              testID="title"
            >
              Peruuta
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
